src/App.tsx

```diff
@@ -1,14 +1,53 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
+import { List } from './components/List';
+import { castomDebounce } from './utils/CastomDebounce';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [query, setQuery] = useState('');
+  const [applyQuery, setApplyQuery] = useState('');
+  const [selectedPerson, setSelectedPerson] = useState({
+    ...peopleFromServer[0],
+    name: 'No selected person',
+  });
+  const { name, born, died } = selectedPerson;
+  const debounceDelay = 1000;
+  const applyedQuery = castomDebounce(setApplyQuery, debounceDelay);
+
+  const visiblePeopleFn = () => {
+    if (applyQuery === '') {
+      return [];
+    }
+
+    const peoples = peopleFromServer.filter(el => (
+      el.name.toLocaleLowerCase()
+        .includes(applyQuery.toLocaleLowerCase().trim())
+    ));
+
+    if (peoples.length === 0) {
```

Comment:
People are already a multitude
```suggestion
    if (!people.length) {
```